salt/states/file.py

```diff
@@ -7805,7 +7805,8 @@ def serialize(
         dataset, **serializer_options.get(serializer_name, {})
     )
 
-    contents += "\n"
+    if isinstance(contents, str):
```

Comment:
the previous implementation would attempt to add a new line to the end of a binary plist. which would cause sadness.

Reply:
My guess is that was written using Py2, where binary == str == text. In Py3, binary == bytes and text == str. Definitely a good move here - unless we should have an 
```
else:
   contents += b"\n"
```

Reply:
@waynew Would it make sense to check to see if a new line already exists? I know `plistlib` automatically adds a newline.

Reply:
I'm not familiar with other tools out there, but I would assume we should do whatever the "correct" thing is.

Reply:
I've updated with the recommendation as well as skipping new line concatenation on plists as this will create an invalid plist file.